### ISSUE TITLE
MPRIS: omit values from `{dynamic}`

### ISSUE THREAD

**Issue by redxtech** [Alexays/Waybar#2084]
Would it be reasonable to add the option to omit certain values from the `{dynamic}` tag?

Something like being able to list the names to never show even if they're filled out:
```json
[
    "ignore-dynamic": [
        "album",
        "length"
    ]
]
```

**Comment by chayleaf** [comment]
you can currently do e.g. `"artist-len": 0` to hide artist. Is this enough for you? There's no way to hide position/length as of yet.

**Comment by redxtech** [comment]
> you can currently do e.g. `"artist-len": 0` to hide artist. Is this enough for you? There's no way to hide position/length as of yet.

That seems decent. If it still leaves the separator between the elements visible then it wouldn't be that great, but if the separator is automatically removed if the length is 0 then this would definitely work.

**Comment by chayleaf** [comment]
it is removed

**Comment by redxtech** [comment]
> the separator is removed. Also, there's `dynamic-len` and `dynamic-priority` which allow you to fit as much tags as possible within a limit

That should be perfect then!

Is there anywhere in the wiki that this is documented?

**Comment by chayleaf** [comment]
it's documented on the man page, I'll update the wiki later when I have time

**Comment by chayleaf** [comment]
yes, as I mentioned above, there isn't currently a way to hide length and position, only make it low priority. It's fairly easy to add though.

**Comment by redxtech** [comment]
ahh my bad. thanks!

**Comment by Alexays** [comment]
This is supported now 🙏 The mpris module has a `dynamic-order` option: list exactly the fields you want in `{dynamic}` (any of `title`, `artist`, `album`, `position`, `length`) and anything omitted is hidden, separators included (src/modules/mpris/mpris.cpp lines 101-107 and 271-274). e.g. `"dynamic-order": ["artist", "title"]` drops album/position/length. Closing as implemented — reopen if you hit an edge case.
